### PR TITLE
Add section for location based updates.

### PR DESCRIPTION
**Pull request by patrickvacek**
Fixes https://github.com/uptane/uptane-standard/issues/161.

### CODE REVIEW

File: customizations.md [hunk 1]
```diff
@@ -160,3 +160,8 @@ In choosing whether to send dynamic directions through the custom field of the T
...
+
+
+## Location-based updates
+
+Certain types of updates, like maps, rules-of-the-road, or traffic notifications, are only relevant to vehicles based on their specific location. These location-based updates require that a device is able to report its location in some way. For example, the device could obtain its location by using a GPS sensor and report it as custom metadata in the vehicle version manifest using the "geo:" UI scheme defined in [https://tools.ietf.org/html/rfc5870](RFC 5870). The Image repository would need to include a similar representation of a location in the custom section of the Targets metadata for applicable targets. The Director would then be responsible for identifying devices whose locations match those of targets in the Image repository. If a match is found, the Director SHOULD update its Targets metadata to instruct the relevant devices to install the location-based updates appropriate for their positions.
```
[the remark this answers is not in the document]
Comment (jhdalek55) [reply customizations.md:167]: Just a few very small suggestions in the first sentence.

**Comment by jhdalek55** [comment]
@trishankkarthik @JustinCappos  could one of you review this for the technical content? It's just one paragraph of text.

File: customizations.md [hunk 2]
```diff
@@ -160,3 +160,8 @@ In choosing whether to send dynamic directions through the custom field of the T
...
+
+
+## Location-based updates
+
+Certain types of updates, like maps, rules-of-the-road, or traffic notifications, are only relevant to vehicles within a specific location. These location-based updates require that a device be able to report its location in some way. For example, the device could obtain its location by using a GPS sensor and report it as custom metadata in the vehicle version manifest using the "geo:" UI scheme defined in [https://tools.ietf.org/html/rfc5870](RFC 5870). The Image repository would need to include a similar representation of a location in the custom section of the Targets metadata for applicable targets. The Director would then be responsible for identifying devices whose locations match those of targets in the Image repository. If a match is found, the Director SHOULD update its Targets metadata to instruct the relevant devices to install the location-based updates appropriate for their positions.
```
Comment (trishankatdatadog) [inline customizations.md:167]: This is a good clarification!

Now, should the custom `targets` metadata on the Image repository _also_ include location metadata?

Comment (patrickvacek) [reply customizations.md:167]: > Now, should the custom `targets` metadata on the Image repository _also_ include location metadata?

Yes, I think it has to. That's what I meant with "The Image repository would need to include a similar representation of a location in the custom section of the Targets metadata for applicable targets." Does that make sense or can I make it clearer?

**Comment by zabbal** [comment]
I like RFC reference. 

Let's double-check that I've understood this correctly:
- imagerepo re-generates targets and includes geo://foo as location-specific updates comes in
- client reports geo://bar in VVM
- director checks that geo://bar is within geo://foo and updates targets metadata by adding geo://foo for client's ECU-id bumping metadata version
- client follows standard verification procedure and also checks for custom geo:// metadata
- once the standard verification passed, the targtes is accepted by client (new version is stored)
- if the geo://foo is still matches the vehicle location, the update is installed, otherwise it's ignored

Am I interpreting this right? Could there be confusion/misunderstanding due to vehicle position change between the time VVM is reported and the time when new targets is received?

Overall, I think it's worth merging as-is already and if further clarifications would be necessary than we can make it in follow-up PRs.

**Comment by patrickvacek** [comment]
> Let's double-check that I've understood this correctly:

Sounds about right to me.

> Could there be confusion/misunderstanding due to vehicle position change between the time VVM is reported and the time when new targets is received?

That is a good question. I hadn't considered that. I think that might be an implementation-specific question, but I could see adding some suggestions here if you think that's wise.

File: customizations.md (hunk 2 above)
Comment (trishankatdatadog) [reply customizations.md:167]: I think it's clearer when you pointed it out, but it doesn't stand out. Maybe Lois could help with that?

Comment (jhdalek55) [reply customizations.md:167]: Perhaps a few small wording tweaks. Does this address your concern @trishankatdatadog? 

```suggestion
A way to reference location in the custom section of the Targets metadata for the Image repository would also be required for all applicable targets. The Director would then be responsible for identifying devices whose locations match those of targets in the Image repository. If a match is found, the Director SHOULD update its Targets metadata to instruct the relevant devices to install the location-based updates appropriate for their positions.
```

Comment (trishankatdatadog) [reply customizations.md:167]: Awesome, thanks, Lois!

**Comment by trishankatdatadog** [comment]
@patrickvacek Yes, if you could add some suggestions that @zabbal recommends, that would be great.

**Comment by patrickvacek** [comment]
Thanks for all the input! I've updated the PR to include Lois's suggestion as well as some considerations about what to do if the location on the device changes as Max brought up.

**Comment by jhdalek55** [comment]
@trishankatdatadog  can you take another look now?

File: customizations.md [hunk 3]
```diff
@@ -160,3 +160,12 @@ In choosing whether to send dynamic directions through the custom field of the T
...
+
+
+## Location-based updates
+
+Certain types of updates, like maps, rules-of-the-road, or traffic notifications, are only relevant to vehicles within a specific location. These location-based updates require that a device be able to report its location in some way. For example, the device could obtain its location by using a GPS sensor and report it as custom metadata in the vehicle version manifest using the "geo:" UI scheme defined in [https://tools.ietf.org/html/rfc5870](RFC 5870).
+
+A way to reference location in the custom section of the Targets metadata for the Image repository would also be required for all applicable targets. The Director would then be responsible for identifying devices whose locations match those of targets in the Image repository. If a match is found, the Director SHOULD update its Targets metadata to instruct the relevant devices to install the location-based updates appropriate for their positions.
```
Comment (trishankatdatadog) [inline customizations.md:169]:
```suggestion
A way to reference location in the custom section of the Targets metadata for the Image repository would then be required for all applicable targets. The Director would then be responsible for identifying devices whose locations match those of targets in the Image repository. If a match is found, the Director SHOULD update its Targets metadata to instruct the relevant devices to install the location-based updates appropriate for their positions.
```

**Comment by trishankatdatadog** [comment]
Approved. I have just one suggestion, and then we are good to go.

**Comment by patrickvacek** [comment]
> I have just one suggestion, and then we are good to go.

Done. Thanks!

**Comment by jhdalek55** [comment]
Ok. Then, can I merge it?

**Comment by trishankatdatadog** [comment]
> Ok. Then, can I merge it?

Feel free, Lois 🙂

**Pull request merged**